src/openvpn/multi.c

```diff
@@ -2481,6 +2486,37 @@ multi_process_float(struct multi_context *m, struct multi_instance *mi)
     gc_free(&gc);
 }
 
+#ifdef ENABLE_VLAN_TAGGING
+/*
+ * Decides whether or not to drop an ethernet frame.  VLAN-tagged frames are
+ * dropped.  All other frames are accepted.
+ *
+ * @param buf The ethernet frame.
+ * @return    Returns true if the frame should be dropped, false otherwise.
+ */
+static bool
+buf_filter_incoming_8021q_vlan_tag (const struct buffer *buf)
```

Comment:
The name is a bit misleading - it doesn't filter anything, it checks whether the frame is a vlan tagged frame. (It was meant as a "policy function", containing the policy decision for when to drop frames.)

Instead I propose the name `vlan_buf_contains_tagged_frame` and would move the log statement out of the function to the single call-site, where the actual dropping happens.

Reply:
This is one of the things I've already refactored into [a vlan.c file](https://github.com/ikelos/openvpn/blob/vlan-patches-v3/src/openvpn/vlan.c#L214).  If it was supposed to be a policy function, then I'd suggest filter is more accurate than what sounds like a tagged/not-tagged test?  It might just be worth bulking out the documentation text around it instead?